### PR TITLE
fix(sdk): Disable concurrency on rewrap

### PR DESCRIPTION
**Pull request by dmihalcik-virtru**
- Adds new `concurrencyLimit` decrypt param, which sets a thread pool (kinda)
- Defaults value to 1

### CODE REVIEW

File: lib/tdf3/src/tdf.ts [hunk 1]
```diff
@@ -982,77 +991,44 @@ async function unwrapKey({
     };
   }
 
-  // Get unique split IDs to determine if we have an OR or AND condition
-  const splitIds = new Set(Object.keys(splitPotentials));
-
-  // If we have only one split ID, it's an OR condition
-  if (splitIds.size === 1) {
-    const [splitId] = splitIds;
+  const poolSize = concurrencyLimit === undefined ? 1 : concurrencyLimit > 1 ? concurrencyLimit : 1;
```
Comment (biscoe916) [inline lib/tdf3/src/tdf.ts:994]: Can we break this down so it's more readable? Nested ternary's are too hard to parse, in my opinion.

Comment (dmihalcik-virtru) [reply lib/tdf3/src/tdf.ts]: better?
